### PR TITLE
Support repeatable directives

### LINKED ISSUE #608: Support `repeatable` directives
The specification recently included a new keyword `repeatable` that allows specifying directives that may be defined multiple times.

https://github.com/graphql/graphql-spec/pull/472

Reading multiple directives already works fine, so that part should not be too hard. We will have to add support for the new keyword and adapt the validation steps to handle it.

Partial support was already added to the reference implementation: https://github.com/graphql/graphql-js/pull/1965

Full support is underway, too: https://github.com/graphql/graphql-js/pull/1541

### PR DESCRIPTION
**Pull request by spawnia**
Resolves #608 

See https://github.com/graphql/graphql-js/pull/1965

### CODE REVIEW

**Comment by spawnia** [comment]
@simPod have you had a chance to review this PR yet?

**Comment by simPod** [comment]
@spawnia I'll take a look but definitely don't feel competent to merge this after on my own, @vladar will have to do it :)

File: src/Validator/Rules/UniqueDirectivesPerLocation.php [hunk 1]
```diff
@@ -26,16 +34,41 @@ public function getSDLVisitor(SDLValidationContext $context)
 
     public function getASTVisitor(ASTValidationContext $context)
     {
+        $uniqueDirectiveMap = [];
+
+        $schema            = $context->getSchema();
+        $definedDirectives = $schema !== null
```
Comment (simPod) [inline src/Validator/Rules/UniqueDirectivesPerLocation.php:40]: Personally, I'd prefer not to use negation in conditions where eligible. Reduces cognitive load while reading it.

File: tests/Type/DirectiveTest.php [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+declare(strict_types=1);
+
+namespace GraphQL\Tests\Type;
+
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @see describe('Type System: Directive', () => {
+ */
+class DirectiveTest extends TestCase
```
Comment (simPod) [inline tests/Type/DirectiveTest.php:12]:
```suggestion
final class DirectiveTest extends TestCase
```

File: src/Validator/Rules/UniqueDirectivesPerLocation.php (hunk 1 above)
Comment (spawnia) [reply src/Validator/Rules/UniqueDirectivesPerLocation.php:40]: Generally agree, although in this case `$schema` is usually non-null. It makes sense to have the happy path first and the fallback afterwards.

Comment (shmax) [reply src/Validator/Rules/UniqueDirectivesPerLocation.php:40]: I will agree with whomever merges some of my PRs first, so I don't have to generate my basefiles again...

**Comment by vladar** [comment]
@spawnia Is it ready for merging? (just to make sure)

**Comment by spawnia** [comment]
> @spawnia Is it ready for merging? (just to make sure)

Yes. I just checked if there are breaking changes we might want to mention, haven't found any.

**Pull request merged**